src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java

```diff
@@ -465,8 +465,16 @@ else if (groupIdx > lastGroup) {
      * @return true if imports group are separated internally.
      */
     private boolean checkSeparatorInGroup(int groupIdx, boolean isStatic, int line) {
-        return !beforeFirstImport && separated && groupIdx == lastGroup
-                && isStatic == lastImportStatic && line - lastImportLine > 1;
+        final boolean isStaticImportGroupIndependent =
+             option == ImportOrderOption.TOP
+                || option == ImportOrderOption.BOTTOM;
+        final boolean inSameGroup =
+            //in same import group
+            groupIdx == lastGroup && isStatic == lastImportStatic
+                //in same static import group
+                || isStaticImportGroupIndependent && isStatic && lastImportStatic;
+        return !beforeFirstImport
+            && inSameGroup && line - lastImportLine > 1;
```

Comment:
as you start to write comment - it is sign that smth wrong id in design.

please rebame method to `isSameGroup(groupIdx, isStatic)`.

and place where it is called should be like:
```
        if (!beforeFirstImport && line - lastImportLine > 1
                && checkSeparatorInGroup(groupIdx, isStatic, line)) {
```

to keep all that mess with `beforeFirstImport` and `lastImportLine` in one place.

Reply:
Done.